unit_tests/test_plugins.py

```diff
@@ -189,6 +189,25 @@ def test_install_bad_plugin(self, mock_restart_jenkins):
         finally:
             hookenv.config()["remove-unlisted-plugins"] = orig_remove_unlisted_plugins
 
+    @mock.patch("test_plugins.Plugins._download_plugin")
+    @mock.patch("test_plugins.Plugins._get_latest_version")
+    @mock.patch("charms.layer.jenkins.api.Api.get_plugin_version")
+    @mock.patch("test_plugins.Plugins._get_plugins_to_install")
+    def test_install_fail(self, mock_get_plugins_to_install, mock_get_plugin_version, mock_get_latest_version, mock_download_plugin, mock_restart_jenkins):
+        """
+        If a plugin is already installed, it doesn't get downloaded.
+        """
```

Comment:
This can be collapsed to:

`"""If a plugin is already installed, it doesn't get downloaded."""`